linchpin/shell/__init__.py

```diff
@@ -336,8 +340,23 @@ def fetch(ctx, fetch_type, remote, root):
               metavar='TX_ID', required=False,
               help="Display a specific transaction by ID (tx_id)."
                    " Only works with '--view=tx'")
+@click.option('--output-format', type=str, default="cfg",
+              metavar='output_format', required=False,
+              help="Inventory output format")
+@click.option('--output-file', type=str,
+              metavar='output_file', required=False,
+              help="inventory output file path")
```

Comment:
I have pointed this out a couple of times, but `--output-file` should be combined with the general option `--output-pinfile` in the general options. You can combine these two options to have something like:

```
@click.option('-o', '--output-file', '--output-pinfile', metavar='OUTPUT_PINFILE',
              help='Write out PinFile to provided location')
```
This way, we continue to use DRY pattern. This goes for combining `--output-format` and `--inventory-format` as well, which I pointed out elsewhere in this review.

Reply:
@herlo I tried combining as you mentioned however , click is not allowing multiple aliases to same option.
```(env_vars) [srallaba@laptopy workspace]$ linchpin journal -t 153 --output-type inventory --output-file /tmp/testinv.txt --output-format json
Traceback (most recent call last):
  File "/home/srallaba/.virtualenvs/env_vars/bin/linchpin", line 11, in <module>
    load_entry_point('linchpin', 'console_scripts', 'linchpin')()
  File "/home/srallaba/.virtualenvs/env_vars/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/srallaba/.virtualenvs/env_vars/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/srallaba/.virtualenvs/env_vars/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/srallaba/.virtualenvs/env_vars/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/srallaba/.virtualenvs/env_vars/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/srallaba/.virtualenvs/env_vars/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/home/srallaba/.virtualenvs/env_vars/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
TypeError: journal() got an unexpected keyword argument 'output_pinfile'
```
The moment we add --output-pinfile as option as follows the above error trace occurs . 
```python
@click.option('-o', '--output-file', '--output-pinfile', type=str,
              metavar='OUTPUT_FILE', required=False,
              help="Write output-file to provided location")
@click.option('--output-type', type=str,
              metavar='output_type', required=False,
              help="inventory output file path")
@click.option('--target', metavar='target', default="all",
              help='If multiple targets are mentioned \
                    takes parameter for target to be used.\
                    By default all are displayed\
                    displayed')
@pass_context
def journal(ctx, targets, fields, count, view,
            tx_id, output_format, output_file, output_type, target):
```

Reply:
@samvarankashyap I was able to make this work without any other modifications. I'm confused as to why you are getting that error on your end. Here's the changes I made.

```
$ git diff linchpin/shell/__init__.py
diff --git a/linchpin/shell/__init__.py b/linchpin/shell/__init__.py
index 83a6e8a..6b67047 100644
--- a/linchpin/shell/__init__.py
+++ b/linchpin/shell/__init__.py
@@ -114,7 +114,8 @@ def _handle_results(ctx, results, return_code):
                    "If template data is from a file, it must be"
                    " prepended with an '@' character."
               )
-@click.option('-o', '--output-pinfile', metavar='OUTPUT_PINFILE',
+@click.option('-o', '--output-file', '--output-pinfile', 'outfile',
+              metavar='OUTPUT_FILE',
               help='Write out PinFile to provided location')
 @click.option('-w', '--workspace', type=click.Path(), envvar='WORKSPACE',
               help='Use the specified workspace. Also works if the'
@@ -127,7 +128,7 @@ def _handle_results(ctx, results, return_code):
               help='Use the specified credentials path. Also works'
                    ' if CREDS_PATH environment variable is set')
 @pass_context
-def runcli(ctx, config, pinfile, template_data, output_pinfile,
+def runcli(ctx, config, pinfile, template_data, outfile,
            workspace, verbose, version, creds_path):
     """linchpin: hybrid cloud orchestration"""
 
@@ -141,8 +142,8 @@ def runcli(ctx, config, pinfile, template_data, output_pinfile,
     # if the pinfile is a template, data will be passed here
     ctx.pf_data = template_data
 
-    if output_pinfile:
-        ctx.set_cfg('tmp', 'output_pinfile', output_pinfile)
+    if outfile:
+        ctx.set_cfg('tmp', 'output_pinfile', outfile)
 
     ctx.pinfile = None
```
When I run `linchpin journal`, or `linchpin up`, I get the expected output. Even the PinFile in JSON format is output when using the `linchpin up` command.

```
$ linchpin -o x journal dummy-new

Target: dummy-new
run_id	    action	     uhash	        rc	
--------------------------------------------------
553    	       up	   1278c4	        0	
552    	       up	   475870	        0	
551    	       up	   f57b00	        0	

$ linchpin --output-file x journal dummy-new

Target: dummy-new
run_id	    action	     uhash	        rc	
--------------------------------------------------
553    	       up	   1278c4	        0	
552    	       up	   475870	        0	
551    	       up	   f57b00	        0	

$ linchpin --output-pinfile x journal dummy-new
 
Target: dummy-new
run_id	    action	     uhash	        rc	
--------------------------------------------------
553    	       up	   1278c4	        0	
552    	       up	   475870	        0	
551    	       up	   f57b00	        0	

$ linchpin -o x up dummy-newAction 'up' on Target 'dummy-new' is complete
-------
start hook shell:hello
hello

end hook shell:hello
-------

ID: 963
Action: up

Target              	Run ID	uHash	Exit Code 
-------------------------------------------------
dummy-new           	   554	39bd76	        0

```
The output files don't work yet on the journal bit (as I didn't implement that logic to test this functionality), but it should not be difficult to make work.

Reply:
@herlo : I see where i am going wrong . I was trying to change journal cli instead of runcli command. Fixing it right now .